### PR TITLE
[dbuf] Distributed Buffer

### PR DESCRIPTION
**Pull request by danking**
I had to recreate from #7593 because I force-pushed after are rebase.

cc: @cseed 

Unfortunately, you're the only one around to review John.

There's so many issues with this at current, but I think it would be better to get something in so I can actually start making forward progress towards something better. At a very practical level, I want various docker images to be cached instead of constantly rebuilding them as I try to develop this further.

### CODE REVIEW

**Comment by danking** [comment]
And some unresolved conversations you should check in on https://github.com/hail-is/hail/pull/7593#discussion_r356485254, https://github.com/hail-is/hail/pull/7593#discussion_r356486036, https://github.com/hail-is/hail/pull/7593#discussion_r356489173

Regarding the last discussion, yeah, I mean the whole distributed thing is completely fucked right now. I just want this in so I can start working on something that sucks less.

File: dbuf/dbuf/client.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    async def get(self, key):
+        server = key[0]
+        server_url = self.deploy_config.base_url(server)
+
+        resp = await utils.request_retry_transient_errors(
+            self.aiosession,
+            'POST',
+            f'{server_url}/s/{self.id}/get',
+            json=key)
+        assert resp.status == 200
```
Comment (johnc1231) [inline dbuf/dbuf/client.py]: As Cotton said, this is still unnecessary in presence of `raise_for_status = True` above.

File: dbuf/dbuf/client.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        self.sizes = []
+        self.cursor = 0
+        self.max_bufsize = max_bufsize
+
+    async def create(self):
+        resp = await utils.request_retry_transient_errors(
+            self.aiosession,
+            'POST',
+            f'{self.root_url}/s')
+        assert resp.status == 200
```
Comment (johnc1231) [inline dbuf/dbuf/client.py]: Not needed

**Comment by danking** [comment]
@johnc1231 addressed.

File: dbuf/dbuf/client.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+        def get_from_server(server, keys):
+            async def f():
+                server_url = self.deploy_config.base_url(server)
+                i = 0
+                while i < len(keys):
+                    batch = []
+                    size = 0
+                    while i < len(keys):
+                        assert keys[i][0][3] < self.max_bufsize
```
Comment (johnc1231) [inline dbuf/dbuf/client.py]: Nit. Personally, I find `keys[i][0][3]` a little hard to read. Like, I know it's the 3rd field of the 0th field of the ith key, but I wish that we used some kind of name or function to access the field that indicated to me what was stored there so I didn't have to go looking around for it.

**Pull request merged**